cabal-testsuite/PackageTests/NewHaddock/DisableDoc/cabal.test.hs

```diff
@@ -0,0 +1,5 @@
+import Test.Cabal.Prelude
+-- Test that `cabal haddock --disable-documention` works as expected and leads
+-- to a warning if a local package makes an outer reference.
```

Comment:
Could we perhaps extend the test to also display the warning (for some other module?)? The very text of the warning may be fragile, but if we notice we need to update it too often as haddock gets message overhauls, we can grep for a couple of keywords instead.

Reply:
Good point. See the `assert` below. Is it good?

Reply:
Oh, I see you approved, so I guess it's good.